plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -1421,8 +1426,8 @@ suggestNewOrExtendImportForClassMethod packageExportsMap ps fileContents Diagnos
             | otherwise -> []
 #endif
 
-suggestNewImport :: ExportsMap -> Annotated ParsedSource -> T.Text -> Diagnostic -> [(T.Text, CodeActionKind, TextEdit)]
-suggestNewImport packageExportsMap ps fileContents Diagnostic{_message}
+suggestNewImport :: DynFlags -> ExportsMap -> Annotated ParsedSource -> T.Text -> Diagnostic -> [(T.Text, CodeActionKind, TextEdit)]
```

Comment:
I, personally, dislike passing around `DynFlags` since they are GHC internals subject to a lot of change. Also, contain a plethora of information unimportant to us.

Are you opposed replacing this with a type, for example:
```haskell
data QualifiedImportStyle = QualifiedImportPostfix | QualifiedImportPrefix
```

and calling functions such as `suggestNewImport` with the `QualifiedImportStyle`?
Name subject to bikeshedding.

Reply:
Sorry for the delay. I've added `QualifiedImportStyle` but kept passing `DynFlags` to `suggestNewImport` as the other options (add `QualifiedImportStyle` to `CodeActionArgs` or somehow compute it inside `iePluginDescriptor` and pass it along) don't seem ideal. I think using the `ToCodeAction` instance for `DynFlags` here is enough?

Reply:
Fair enough, SGTM!